stl/inc/expected

```diff
@@ -0,0 +1,1022 @@
+    // clang-format on
+    static_assert(!_Is_specialization_v<_Err, unexpected>, "E must not be a specialization of unexpected");
+
+    template <class _Ty, copy_constructible _Err2> // TRANSITION, LWG-3688
+    friend class expected;
+
+public:
+    // [expected.un.ctor]
+    template <class... _Args>
+        requires is_constructible_v<_Err, _Args...>
+    constexpr explicit unexpected(in_place_t, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_Err, _Args...>) // strengthened
+        : _Unexpected(_STD forward<_Args>(_Vals)...) {}
+
+    // clang-format off
+    template <class _Uty, class... _Args>
+        requires is_constructible_v<_Err, initializer_list<_Uty>&, _Args...>
+    constexpr explicit unexpected(in_place_t, initializer_list<_Uty> _Ilist, _Args&&... _Vals) noexcept(
+               is_nothrow_constructible_v<_Err, initializer_list<_Uty>&, _Args...>) // strengthened
+        : _Unexpected(_Ilist, _STD forward<_Args>(_Vals)...) {}
+
+    template <class _UError = _Err>
+        requires (!is_same_v<remove_cvref_t<_UError>, unexpected> && !is_same_v<remove_cvref_t<_UError>, in_place_t> //
+                 && is_constructible_v<_Err, _UError>)
+    constexpr explicit unexpected(_UError&& _Unex) noexcept(is_nothrow_constructible_v<_Err, _UError>) // strengthened
+        : _Unexpected(_STD forward<_UError>(_Unex)) {}
+    // clang-format on
+
+    // [expected.un.observe]
+#ifdef __cpp_explicit_this_parameter // TRANSITION, P2549
+    _NODISCARD constexpr decltype(auto) error(this unexpected&& _Self) noexcept {
+        return _STD forward<unexpected>(_Self)._Unexpected;
+    }
+#else // ^^^ __cpp_explicit_this_parameter ^^^ / vvv !__cpp_explicit_this_parameter vvv
+    _NODISCARD constexpr const _Err& error() const& noexcept {
+        return _Unexpected;
+    }
+    _NODISCARD constexpr _Err& error() & noexcept {
+        return _Unexpected;
+    }
+    _NODISCARD constexpr const _Err&& error() const&& noexcept {
+        return _STD move(_Unexpected);
+    }
+    _NODISCARD constexpr _Err&& error() && noexcept {
+        return _STD move(_Unexpected);
+    }
+#endif // !__cpp_explicit_this_parameter
+
+    // [expected.un.swap]
+    constexpr void swap(unexpected& _Other) noexcept(is_nothrow_swappable_v<_Err>) requires is_swappable_v<_Err> {
+        _Swap_adl(_Unexpected, _Other._Unexpected);
+    }
+
+    friend constexpr void swap(unexpected& _Left, unexpected& _Right) noexcept(
+        is_nothrow_swappable_v<_Err>) requires is_swappable_v<_Err> {
+        _Left.swap(_Right);
+    }
+
+    // [expected.un.eq]
+    template <_Weakly_equality_comparable_with<_Err> _UErr>
+    _NODISCARD_FRIEND constexpr bool operator==(const unexpected& _Left, const unexpected<_UErr>& _Right) noexcept( //
+        noexcept(_Implicitly_convert_to<bool>(_Left._Unexpected == _Right.error()))) { // strengthened
+        return _Left._Unexpected == _Right.error();
+    }
+
+private:
+    _Err _Unexpected;
+};

```

Comment:
The deduction guide `template<class E> unexpected(E) -> unexpected<E>;` depicted by [expected.un.object.general] is missing. This also indicates missing test coverage.